tracer/src/Datadog.Trace/RemoteConfigurationManagement/RemoteConfigurationManager.cs

```diff
@@ -222,11 +247,26 @@ private GetRcmRequest BuildRequest(IDictionary<string, Product> products)
 
             var rcmState = new RcmClientState(_rootVersion, _targetsVersion, configStates, _lastPollError != null, _lastPollError, _backendClientState);
             var rcmClient = new RcmClient(_id, products.Keys, _rcmTracer, rcmState, capabilitiesArray);
+            EnrichWithSourceLinkTags(rcmClient.ClientTracer.Tags);
             var rcmRequest = new GetRcmRequest(rcmClient, cachedTargetFiles);
 
             return rcmRequest;
         }
 
+        private void EnrichWithSourceLinkTags(List<string> tags)
+        {
+            if (tags.Any(t => t.StartsWith(CommonTags.GitTag)) && tags.Any(t => t.StartsWith(CommonTags.GitRepository)))
+            {
+                // We already have both tags, no need to add more
+                return;
+            }
+
+            if (SourceLinkTagsProvider.Instance.GetGitTagsFromSourceLink() is { } gitTags)
```

Comment:
Same arguments as before about calling this multiple times - once it's successful we can presumably avoid calling it again

Reply:
Fixed as part of 52d6dff455366205f331e1c766c27cb86804cc07